src/runtime/relax_vm/cuda/cuda_graph_builtin.cc

```diff
@@ -31,6 +31,46 @@ namespace tvm {
 namespace runtime {
 namespace relax_vm {
 
+struct CUDAGraphCaptureKey {
+  int64_t index;
+  ShapeTuple shape_expr;
+
+  CUDAGraphCaptureKey(int64_t index, const Optional<ShapeTuple>& shape_expr) : index(index) {
+    if (shape_expr) {
+      this->shape_expr = shape_expr.value();
+    }
+  }
+
+  bool operator==(const CUDAGraphCaptureKey& other) const {
+    return index == other.index && std::equal(shape_expr.begin(), shape_expr.end(),
+                                              other.shape_expr.begin(), other.shape_expr.end());
```

Comment:
need to compare the shape_expr.size not null first, then shape expr size then content

Reply:
this overload of `std::equal` check the two range of the same size

Reply:
what happens if shape_expr is nullptr?